pages/marriage.php

```diff
@@ -148,10 +148,49 @@ function marriage() {
 
     if(isset($_POST['confirm_divorce'])) {
         try {
-            $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`='$player->spouse' LIMIT 1");
-            if(!$system->db_last_affected_rows) {
-                throw new Exception("Error processing divorce!");
+            $result = $system->query("SELECT user_id, user_name, spouse FROM users WHERE user_id ='$player->spouse' LIMIT 1");
+            if (!$system->db_last_num_rows)
+            {
+                throw new Exception("Failed to fetch marriage information");
```

Comment:
I think in this case we should clear the player's spouse, otherwise we won't solve the one-sided marriage when the other person isn't married